x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/role_selector.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { startCase } from 'lodash';
+
+import {
+  EuiCallOut,
+  EuiFormRow,
+  EuiRadio,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+  EuiTitle,
+} from '@elastic/eui';
+
+interface IRoleSelectorProps {
+  disabled?: boolean;
+  disabledText?: string;
+  roleType?: string;
+  roleTypeOption: string;
+  description: string;
+  onChange(roleTypeOption: string): void;
+}
+
+export const RoleSelector: React.FC<IRoleSelectorProps> = ({
+  disabled,
+  disabledText,
+  roleType,
+  roleTypeOption,
+  description,
+  onChange,
+}) => (
+  <EuiFormRow>
+    <EuiRadio
+      disabled={disabled}
+      id={roleTypeOption}
+      checked={roleTypeOption === roleType}
```

Comment:
A test to check that `checked` is passed as true when `roleTypeOption === roleType` would be good to have.

Reply:
You could stuff like this in the `renders` test...

```tsx
  it('renders', () => {
    const wrapper = shallow(<RoleSelector {...props} />);

    expect(wrapper.find(EuiRadio).prop('disabled')).toEqual(props.disabled);
    expect(wrapper.find(EuiRadio).prop('checked')).toEqual(true);
  });
```

And then an additional test for the negative case for checked
```tsx
  it('renders as unchecked of roleTypeOption does not equal roleType', () => {
    const wrapper = shallow(<RoleSelector {...{
      ...props,
      roleType: 'foo'
    }} />);

    expect(wrapper.find(EuiRadio).prop('checked')).toEqual(false);
  });
```